boards/lilygo/ttgo_t8s3/Kconfig.defconfig

```diff
@@ -0,0 +1,19 @@
+# TTGO T8S3 board configuration
+
+# Copyright (c) 2024 Lothar Felten <lothar.felten@gmail.com>
+# SPDX-License-Identifier: Apache-2.0
+
+if BOARD_TTGO_T8S3_ESP32S3_PROCPU
+
+config HEAP_MEM_POOL_ADD_SIZE_BOARD
+	int
+	default 4096
+
+endif # BOARD_TTGO_T8S3_ESP32S3_PROCPU
+
+if BOARD_TTGO_T8S3_ESP32S3_APPCPU
+
+config HEAP_MEM_POOL_ADD_SIZE_BOARD
+	default 256
+
+endif # BOARD_TTGO_T8S3_ESP32S3_APPCPU
```

Comment:
this needs to go in Kconfig as:
```
config HEAP_MEM_POOL_ADD_SIZE_BOARD
	int
	default 4096 if BOARD_TTGO_T8S3_ESP32S3_PROCPU
	default 256 if BOARD_TTGO_T8S3_ESP32S3_APPCPU
```

Reply:
currently I  have no Kconfig without a suffix.
I noticed some of the nordic boards have that, but none of the espressif ones. 
Is this new structure the way to go?

Reply:
Kconfig is correct, Kconfig.defconfig is only for setting defaults on existing values. @sylvioalves all of the following need to be fixed:
```
./01space/esp32c3_042_oled/Kconfig.defconfig:config HEAP_MEM_POOL_ADD_SIZE_BOARD
./dptechnics/walter/Kconfig.defconfig:config HEAP_MEM_POOL_ADD_SIZE_BOARD
./dptechnics/walter/Kconfig.defconfig:config HEAP_MEM_POOL_ADD_SIZE_BOARD
./espressif/esp32_devkitc_wroom/Kconfig.defconfig:config HEAP_MEM_POOL_ADD_SIZE_BOARD
./espressif/esp32_devkitc_wroom/Kconfig.defconfig:config HEAP_MEM_POOL_ADD_SIZE_BOARD
./espressif/esp32_devkitc_wrover/Kconfig.defconfig:config HEAP_MEM_POOL_ADD_SIZE_BOARD
./espressif/esp32_devkitc_wrover/Kconfig.defconfig:config HEAP_MEM_POOL_ADD_SIZE_BOARD
./espressif/esp32_ethernet_kit/Kconfig.defconfig:config HEAP_MEM_POOL_ADD_SIZE_BOARD
./espressif/esp32_ethernet_kit/Kconfig.defconfig:config HEAP_MEM_POOL_ADD_SIZE_BOARD
./espressif/esp32c3_devkitc/Kconfig.defconfig:config HEAP_MEM_POOL_ADD_SIZE_BOARD
./espressif/esp32c3_devkitm/Kconfig.defconfig:config HEAP_MEM_POOL_ADD_SIZE_BOARD
./espressif/esp32c3_rust/Kconfig.defconfig:config HEAP_MEM_POOL_ADD_SIZE_BOARD
./espressif/esp32c6_devkitc/Kconfig.defconfig:config HEAP_MEM_POOL_ADD_SIZE_BOARD
./espressif/esp32s2_devkitc/Kconfig.defconfig:config HEAP_MEM_POOL_ADD_SIZE_BOARD
./espressif/esp32s2_saola/Kconfig.defconfig:config HEAP_MEM_POOL_ADD_SIZE_BOARD
./espressif/esp32s3_devkitc/Kconfig.defconfig:config HEAP_MEM_POOL_ADD_SIZE_BOARD
./espressif/esp32s3_devkitc/Kconfig.defconfig:config HEAP_MEM_POOL_ADD_SIZE_BOARD
./espressif/esp32s3_devkitm/Kconfig.defconfig:config HEAP_MEM_POOL_ADD_SIZE_BOARD
./espressif/esp32s3_devkitm/Kconfig.defconfig:config HEAP_MEM_POOL_ADD_SIZE_BOARD
./espressif/esp32s3_eye/Kconfig.defconfig:config HEAP_MEM_POOL_ADD_SIZE_BOARD
./espressif/esp32s3_eye/Kconfig.defconfig:config HEAP_MEM_POOL_ADD_SIZE_BOARD
./espressif/esp8684_devkitm/Kconfig.defconfig:config HEAP_MEM_POOL_ADD_SIZE_BOARD
./espressif/esp_wrover_kit/Kconfig.defconfig:config HEAP_MEM_POOL_ADD_SIZE_BOARD
./espressif/esp_wrover_kit/Kconfig.defconfig:config HEAP_MEM_POOL_ADD_SIZE_BOARD
./franzininho/esp32s2_franzininho/Kconfig.defconfig:config HEAP_MEM_POOL_ADD_SIZE_BOARD
./hardkernel/odroid_go/Kconfig.defconfig:config HEAP_MEM_POOL_ADD_SIZE_BOARD
./hardkernel/odroid_go/Kconfig.defconfig:config HEAP_MEM_POOL_ADD_SIZE_BOARD
./heltec/heltec_wifi_lora32_v2/Kconfig.defconfig:config HEAP_MEM_POOL_ADD_SIZE_BOARD
./heltec/heltec_wifi_lora32_v2/Kconfig.defconfig:config HEAP_MEM_POOL_ADD_SIZE_BOARD
./heltec/heltec_wireless_stick_lite_v3/Kconfig.defconfig:config HEAP_MEM_POOL_ADD_SIZE_BOARD
./heltec/heltec_wireless_stick_lite_v3/Kconfig.defconfig:config HEAP_MEM_POOL_ADD_SIZE_BOARD
./infineon/cy8cproto_062_4343w/Kconfig.defconfig:config HEAP_MEM_POOL_ADD_SIZE_BOARD
./infineon/cyw920829m2evk_02/Kconfig.defconfig:config HEAP_MEM_POOL_ADD_SIZE_BOARD
./kincony/kincony_kc868_a32/Kconfig.defconfig:config HEAP_MEM_POOL_ADD_SIZE_BOARD
./kincony/kincony_kc868_a32/Kconfig.defconfig:config HEAP_MEM_POOL_ADD_SIZE_BOARD
./lilygo/ttgo_lora32/Kconfig.defconfig:config HEAP_MEM_POOL_ADD_SIZE_BOARD
./lilygo/ttgo_lora32/Kconfig.defconfig:config HEAP_MEM_POOL_ADD_SIZE_BOARD
./lilygo/ttgo_t8c3/Kconfig.defconfig:config HEAP_MEM_POOL_ADD_SIZE_BOARD
./luatos/esp32c3_luatos_core/Kconfig.defconfig:config HEAP_MEM_POOL_ADD_SIZE_BOARD
./luatos/esp32s3_luatos_core/Kconfig.defconfig:config HEAP_MEM_POOL_ADD_SIZE_BOARD
./luatos/esp32s3_luatos_core/Kconfig.defconfig:config HEAP_MEM_POOL_ADD_SIZE_BOARD
./m5stack/m5stack_atom_lite/Kconfig.defconfig:config HEAP_MEM_POOL_ADD_SIZE_BOARD
./m5stack/m5stack_atom_lite/Kconfig.defconfig:config HEAP_MEM_POOL_ADD_SIZE_BOARD
./m5stack/m5stack_atoms3/Kconfig.defconfig:config HEAP_MEM_POOL_ADD_SIZE_BOARD
./m5stack/m5stack_atoms3/Kconfig.defconfig:config HEAP_MEM_POOL_ADD_SIZE_BOARD
./m5stack/m5stack_atoms3_lite/Kconfig.defconfig:config HEAP_MEM_POOL_ADD_SIZE_BOARD
./m5stack/m5stack_atoms3_lite/Kconfig.defconfig:config HEAP_MEM_POOL_ADD_SIZE_BOARD
./m5stack/m5stack_core2/Kconfig.defconfig:config HEAP_MEM_POOL_ADD_SIZE_BOARD
./m5stack/m5stack_core2/Kconfig.defconfig:config HEAP_MEM_POOL_ADD_SIZE_BOARD
./m5stack/m5stack_stamps3/Kconfig.defconfig:config HEAP_MEM_POOL_ADD_SIZE_BOARD
./m5stack/m5stack_stamps3/Kconfig.defconfig:config HEAP_MEM_POOL_ADD_SIZE_BOARD
./m5stack/m5stickc_plus/Kconfig.defconfig:config HEAP_MEM_POOL_ADD_SIZE_BOARD
./m5stack/m5stickc_plus/Kconfig.defconfig:config HEAP_MEM_POOL_ADD_SIZE_BOARD
./m5stack/stamp_c3/Kconfig.defconfig:config HEAP_MEM_POOL_ADD_SIZE_BOARD
./native/nrf_bsim/Kconfig.defconfig:config HEAP_MEM_POOL_ADD_SIZE_BOARD
./olimex/olimex_esp32_evb/Kconfig.defconfig:config HEAP_MEM_POOL_ADD_SIZE_BOARD
./olimex/olimex_esp32_evb/Kconfig.defconfig:config HEAP_MEM_POOL_ADD_SIZE_BOARD
./others/icev_wireless/Kconfig.defconfig:config HEAP_MEM_POOL_ADD_SIZE_BOARD
./seeed/xiao_esp32c3/Kconfig.defconfig:config HEAP_MEM_POOL_ADD_SIZE_BOARD
./seeed/xiao_esp32s3/Kconfig.defconfig:config HEAP_MEM_POOL_ADD_SIZE_BOARD
./seeed/xiao_esp32s3/Kconfig.defconfig:config HEAP_MEM_POOL_ADD_SIZE_BOARD
./vcc-gnd/yd_esp32/Kconfig.defconfig:config HEAP_MEM_POOL_ADD_SIZE_BOARD
./vcc-gnd/yd_esp32/Kconfig.defconfig:config HEAP_MEM_POOL_ADD_SIZE_BOARD
./waveshare/esp32s3_touch_lcd_1_28/Kconfig.defconfig:config HEAP_MEM_POOL_ADD_SIZE_BOARD
./waveshare/esp32s3_touch_lcd_1_28/Kconfig.defconfig:config HEAP_MEM_POOL_ADD_SIZE_BOARD
./wemos/esp32s2_lolin_mini/Kconfig.defconfig:config HEAP_MEM_POOL_ADD_SIZE_BOARD
```

Reply:
@nordicjm The approach used in all those files are the same used in other vendors. It was suggested during hwmv2. I guess then other boards also required that change. Right?

Reply:
Your suggestion is to move those to `kcconfig.board_name`?

Reply:
Any board that put it into Kconfig.defconfig needs to move it out to Kconfig

Reply:
No, `Kconfig`

Reply:
In the case below:
```c
if BOARD_ESP32_ETHERNET_KIT_ESP32_PROCPU

config ESP_SPIRAM
	default y if !MCUBOOT

choice SPIRAM_TYPE
	default SPIRAM_TYPE_ESPPSRAM64
endchoice

config HEAP_MEM_POOL_ADD_SIZE_BOARD
	int
	default 8192

endif # BOARD_ESP32_ETHERNET_KIT_ESP32_PROCPU

if BOARD_ESP32_ETHERNET_KIT_ESP32_APPCPU

config HEAP_MEM_POOL_ADD_SIZE_BOARD
	default 4096

endif # BOARD_ESP32_ETHERNET_KIT_ESP32_APPCPU

```
Should we move everything to `Kconfig` as is?

Reply:
Only `HEAP_MEM_POOL_ADD_SIZE_BOARD` as that is a symbol you define, the other parts are zephyr-defined symbols and you are changing the defaults, so Kconfig.defconfig is the right place for them

Reply:
@nordicjm `Only HEAP_MEM_POOL_ADD_SIZE_BOARD as that is a symbol you define`. What do you mean by that?  `HEAP_MEM_POOL_ADD_SIZE_BOARD` is a zephyr-defined symbol.

Reply:
Hello, I replaced the Kconfig.defconfig with a Kconfig as nordicjm suggested.

Reply:
> Edit: Ok, this is a runtime symbol that is not present in any Kconfig by default. Is that your point after all?

Yes, it's board defined, defconfig is for existing zephyr Kconfigs, you do not put new locally defined Kconfigs in it